### PR TITLE
feat(GameBanana): support game manager pages and other changes

### PR DESCRIPTION
**Pull request by MegAmi24**
## Description 
This update to the GameBanana presence adds support to game manager related pages, including game settings pages and the modlog. The names for withheld and trashed submissions are now hidden, similarly to private submissions. Instances of "a news" have been changed to "a news article", and the strings "Viewing featured for:" and "Viewing rules for:" have been changed to "Viewing featured submissions for:" and "Viewing the rules for:" respectively.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image
image

</details>

### CODE REVIEW

**Comment by ImRodry** [comment]
This change should be semver:minor

**Comment by ImRodry** [comment]
I don’t know what your idea of semver is but it should follow the format MAJOR.MINOR.PATCH. Please fix this

**Comment by MegAmi24** [comment]
I get very confused trying to figure it out because your own guidelines don't give the proper scheme. 
2641A9C1-8564-4D58-9531-13F3EFFF5891

**Comment by ImRodry** [comment]
yeah that's wrong, we follow the regular semver standard

**Comment by ImRodry** [comment]
The commit history on this branch was quite messy so I squashed all commits in order to rebase and fix the issues related to the latest changes in our prettier configuration. If you need help with your local branch, please ping me in #presence-dev on Discord.

File: websites/G/GameBanana/presence.ts [hunk 1]
```diff
@@ -243,12 +292,17 @@ presence.on("UpdateData", async () => {
 				}
 				if (document.querySelector("#PrivateAccessNoticeModule"))
 					presenceData.state = "(Private)";
+				else if (document.querySelector("#WithholdNoticeModule"))
+					presenceData.state = "(Withheld)";
+				else if (document.querySelector("#TrashNoticeModule"))
+					presenceData.state = "(Trashed)";
```
Comment (ImRodry) [inline websites/G/GameBanana/presence.ts:298]: why is the only word in the state in parenthesis?

File: websites/G/GameBanana/presence.ts [hunk 2]
```diff
@@ -60,6 +60,14 @@ presence.on("UpdateData", async () => {
...
 			"Wiki",
+		],
+		gamesettingspages = [
+			"welcomemessage",
+			"subcontrols",
+			"rules",
+			"attribs",
+			"details",
+			"uberstyle",
 		];
```
Comment (ImRodry) [inline websites/G/GameBanana/presence.ts:71]: all of these variables are never modified, you should declare them at the top level to save some performance

File: websites/G/GameBanana/presence.ts [hunk 3]
```diff
@@ -84,12 +92,48 @@ presence.on("UpdateData", async () => {
...
+				).content;
+			} else if (document.location.pathname.includes("/edit")) {
+				presenceData.details = "Editing a game:";
+				presenceData.state = document.querySelector<HTMLMetaElement>(
+					'meta[property="gb:game_name"]'
+				).content;
+			} else if (document.location.pathname.includes("/settings")) {
+				if (
+					gamesettingspages.includes(
+						document.location.pathname.split("settings/")[1].split("/")[0]
```
Comment (ImRodry) [inline websites/G/GameBanana/presence.ts:108]: couldn't you just split at / and adjust the index?

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/G/GameBanana/presence.ts (hunk 2 above)
Comment (MegAmi24) [reply websites/G/GameBanana/presence.ts:71]: What do you mean by "the top level"?

Comment (ImRodry) [reply websites/G/GameBanana/presence.ts:71]: I mean it should be declared in the global scope as opposed to inside the event. And yes, those too

**Pull request closed without merging**